drivers/hmis/app/graphql/types/hmis_schema/service_type_input.rb

```diff
@@ -8,14 +8,26 @@ module Types
   class HmisSchema::ServiceTypeInput < BaseInputObject
     description 'Create service type input'
 
-    argument :name, String, required: true
-    argument :service_category_id, ID, required: true
+    argument :name, String, required: false
+    argument :service_category_id, ID, required: false
+    argument :service_category_name, String, required: false
     argument :supports_bulk_assignment, Boolean, required: false
 
     def to_params
-      result = to_h.except(:service_category_id)
-      result[:custom_service_category_id] = service_category_id
-      result
+      to_h.except(:service_category_id, :service_category_name)
+    end
+
+    def get_or_create_service_category(user_id, data_source_id)
+      # will be nil if neither is provided; that's fine for update (but not for create, which validates presence)
+      if service_category_id.present?
+        Hmis::Hud::CustomServiceCategory.find(service_category_id)
+      elsif service_category_name.present?
+        Hmis::Hud::CustomServiceCategory.new(
```

Comment:
To avoid creating duplicate service categories, lets use an existing service of that name if it exists. Something like:
```
Hmis::Hud::CustomServiceCategory.where(data_source_id: data_source_id, name: service_category_name).first_or_initialize(user_id: user_id)
```